library/pod-security-policy/fsgroup/src_test.rego

```diff
@@ -0,0 +1,124 @@
+package k8spspfsgroup
+
+test_input_fsgroup_allowed_all {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters_runasany}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_no_fsgroup_allowed_all {
+    input := { "review": input_review, "parameters": input_parameters_runasany}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MustRunAs_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters_in_list_mustrunas}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MustRunAs_not_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters_in_list_mustrunas_outofrange}
+    results := violation with input as input
+    count(results) > 0
+}
+test_input_no_fsgroup_MustRunAs_not_allowed {
+    input := { "review": input_review, "parameters": input_parameters_in_list_mustrunas}
+    results := violation with input as input
+    count(results) > 0
+}
+test_input_fsgroup_MayRunAs_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters_in_list_mayrunas}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MayRunAs_not_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters_in_list_mayrunas_outofrange}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_no_fsgroup_MayRunAs_not_allowed {
+    input := { "review": input_review, "parameters": input_parameters_in_list_mayrunas}
+    results := violation with input as input
+    count(results) == 0
+}
+
+input_review = {
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {
+            "containers": input_containers_one,
+            "volumes": input_volumes
+      }
+    }
+}
+
```

Comment:
I'd have a test that includes spec.SecurityContext but not spec.securityContext.fsGroup